### PR TITLE
Remove Twisted from marketplace logic

### PR DESCRIPTION
**Pull request by mplebanski**

### CODE REVIEW

**Comment by mbenke** [comment]
I suggest to add an abstract OfferPool superclass and rename OfferPool to BrassOfferPool or sth

**Comment by mbenke** [comment]
Or, to sync with Architecture nomenclature:

```
RequestorMarketStrategy
BrassRequestorMarketStrategy
WasmRequestorMarketStrategy
```

File: golem/marketplace/offerpool.py [hunk 1]
```diff
@@ -30,46 +33,32 @@ def __init__(
...
-    def change_interval(cls, interval: float) -> None:
-        logger.info("Offer pooling interval set to %.1f", interval)
-        cls._INTERVAL = interval
+    def add(cls, task: Task, provider_id: str, price: float) -> None:
+        if task.header.task_id not in cls._pools:
+            cls._pools[task.header.task_id] = []
+        offer = Offer(
+            scaled_price=scale_price(task.header.max_price, price),
+            reputation=get_provider_efficiency(provider_id),
+            quality=get_provider_efficacy(provider_id).vector,
```
Comment (mbenke) [inline golem/marketplace/offerpool.py]:
```
golem/marketplace/offerpool.py:42: error: Argument "quality" to "Offer" has incompatible type "Tuple[float, ...]"; expected "Tuple[float, float, float, float]"
```

**Comment by mbenke** [comment]
More lint complaints (not our code AFAIK):

```
golem/task/taskserver.py:153: error: Need type annotation for 'last_messages'
golem/task/taskserver.py:155: error: Need type annotation for 'results_to_send'
golem/task/taskserver.py:156: error: Need type annotation for 'failures_to_send'
golem/task/taskserver.py:162: error: Need type annotation for 'forwarded_session_requests'
golem/task/taskserver.py:166: error: Need type annotation for 'resource_handshakes'
golem/task/tasksession.py:617: error: Argument 1 to "task_given" of "TaskServer" has incompatible type "Optional[str]"; expected "str"
```

File: golem/marketplace/offerpool.py [hunk 2]
```diff
@@ -30,46 +33,32 @@ def __init__(
...
 
     @classmethod
-    def _choose_offers(cls, task_id: str) -> None:
-        logger.info("Ordering providers for task: %s", task_id)
-        offers = cls._pools.pop(task_id)
-        order = order_providers(list(map(lambda x: x[0], offers)))
-        for i in order:
-            offers[i][1].callback(True)
+    def get_task_offer_count(cls, task) -> int:
+        return len(cls._pools[task.header.task_id]) if task.header.task_id in cls._pools else 0
```
Comment (mbenke) [inline golem/marketplace/offerpool.py]:
```
C: golem/marketplace/offerpool.py:64:0: Final newline missing (missing-final-newline)
C: golem/marketplace/offerpool.py:64:0: Line too long (95/80) (line-too-long)
C: golem/marketplace/offerpool.py:64:0: Mixed line endings LF and CRLF (mixed-line-endings)
```

File: golem/task/tasksession.py [hunk 3]
```diff
@@ -13,26 +13,25 @@
...
+from twisted.internet.defer import Deferred
 
 import golem
 from golem.core import common
 from golem.core import golem_async
 from golem.core import variables
 from golem.docker.environment import DockerEnvironment
 from golem.docker.image import DockerImage
-from golem.marketplace import scale_price, Offer, OfferPool
+from golem.marketplace import Offer, OfferPool
```
Comment (mbenke) [inline golem/task/tasksession.py]:
```
W: golem/task/tasksession.py:27:0: Unused Offer imported from golem.marketplace (unused-import)
```

**Comment by mbenke** [comment]
```
W: golem/task/tasksession.py:5:0: Unused import functools (unused-import)
```

File: golem/marketplace/offerpool.py [hunk 4]
```diff
@@ -30,46 +33,32 @@ def __init__(
...
+            cls._pools[task.header.task_id] = []
+        offer = Offer(
+            scaled_price=scale_price(task.header.max_price, price),
+            reputation=get_provider_efficiency(provider_id),
+            quality=get_provider_efficacy(provider_id).vector,
+        )
+        cls._pools[task.header.task_id].append(offer)
 
-    @classmethod
-    def add(cls, task_id: str, offer: Offer) -> Deferred:
```
Comment (Krigpl) [inline golem/marketplace/offerpool.py:42]: Do not replace the `Offer` class with `Task`, `Task` is a monstrosity which does a lot more than `OfferPool` requires - iow keep the required interface to the minimum.

Comment (mbenke) [reply golem/marketplace/offerpool.py:42]: @Krigpl We could have `task_id` instead of task, but how to get `max_price` from `task_id`?

**Comment by mbenke** [comment]
Forced build: https://buildbot.golem.network/buildbot/#/builders/9/builds/13449

**Pull request closed without merging**